articles/extensions/delegated-admin.md

```diff
@@ -2,58 +2,89 @@
+Following that, navigate to [Database Connections](${manage_url}/#/connections/database) and click on **+Create DB Connection**. Set a name for your connection, we will name ours *Helpdesk*.
 
 ![](/media/articles/extensions/delegated-admin/create-connection.png)
 
-Navigate to the *Settings* tab of the new connection and enable the **Disable Sign Ups** option. This way we avoid another security concern: if some malicious user gets hold of the link, signing up will not be possible. 
+Navigate to the *Settings* tab of the new connection and enable the **Disable Sign Ups** option. This way we avoid another security concern: if some malicious user gets hold of the link, signing up will not be possible.
 
 ![](/media/articles/extensions/delegated-admin/disable-signup.png)
 
 Enable this new connection for your client (*Users Dashboard* in our case) and add at least one user.
 
+### Assing the right Roles to your Users
+
+Access to the extension requires your users to have the right roles:
+
+ - `Delegated Admin - User`: These users can search for users, create users, open users and execute actions on these users (like Delete, Block ...)
+ - `Delegated Admin - Administrator`: These users can additional see all logs in an account and configure Hooks
+
+Only users that have these roles defined in `user.roles`, `user.app_metadata.roles` or `user.app_metadata.authorization.roles` will be able to use the extension. You could manually set these roles in your users or use a rule in order to do this. The following rule shows how users from the `IT Department` are given the Administrator role while "Department Managers" are given the User role.
+
+```js
+function (user, context, callback) {
+ if (context.clientID === 'CLIENT_ID') {
+   if (user.groups && user.groups.indexOf('IT Department') > -1) {
+     user.roles = user.roles || [ ];
+     user.roles.push('Delegated Admin - Administrator');
+     return callback(null, user, context);
+   } else if (user.app_metadata && user.app_metadata.isDepartmentManager && user.app_metadata.department && user.app_metadata.department.length) {
+     user.roles = user.roles || [ ];
+     user.roles.push('Delegated Admin - User');
+     return callback(null, user, context);
+   }
+
+   return callback(new UnauthorizedError('You are not allowed to use this application.'));
+ }
+
+ callback(null, user, context);
+}
+```
+
+## Install the Extension
+
 We are now ready to setup our new extension. Before we do so head back to your new Client and copy the **Client ID** value.
 
 To install and configure this extension, click on the **Delegated Administration** box in the list of provided extensions on the [Extensions](${manage_url}/#/extensions) page of the dashboard. The **Install Extension** window will open.
 
 ![](/media/articles/extensions/delegated-admin/install-extension.png)
```

Comment:
I didn't update this one, not sure if I should. I can see the same thing, only the version at the top right changes.